common/oneshot_broadcast/src/lib.rs

```diff
@@ -0,0 +1,182 @@
+//! Provides a single-sender, multiple receiver one-shot channel where any message sent will be
+//! received by all senders.
+//!
+//! This implementation may not be blazingly fast but it should be simple enough to be reliable.
+use parking_lot::{Condvar, Mutex};
+use std::sync::{Arc, Weak};
+
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum Error {
+    SenderDropped,
+}
+
+enum Future<T> {
+    /// The future is ready and the item may be consumed.
+    Ready(T),
+    /// Future is not ready. The contained `Weak` is a reference to the `Sender` that may be used to
+    /// detect when the channel is disconnected.
+    NotReady(Weak<()>),
+}
+
+struct MutexCondvar<T> {
+    mutex: Mutex<Future<T>>,
+    condvar: Condvar,
+}
+
+/// The sending pair of the `oneshot` channel.
+pub struct Sender<T>(Arc<MutexCondvar<T>>, Option<Arc<()>>);
```

Comment:
The `Arc<()>` could maybe be an `AtomicBool`, but that would require specifying memory orderings and stuff, so I think the `Arc<()>` is probably more readable (and should be just as efficient at runtime).